components/infobox/wikis/clashofclans/infobox_league_custom.lua

```diff
@@ -0,0 +1,71 @@
+---
+-- @Liquipedia
+-- wiki=clashofclans
+-- page=Module:Infobox/League/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local League = Lua.import('Module:Infobox/League', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+local SUPERCELL_SPONSORED_ICON = '[[File:Supercell lightmode.png|x18px|link=Supercell'
+	.. '|Tournament sponsored by Supercell.|class=show-when-light-mode]][[File:Supercell darkmode.png'
+	.. '|x18px|link=Supercell|Tournament sponsored by Supercell.|class=show-when-dark-mode]]'
+
+local _args
+local _league
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_league = league
+	_args = _league.args
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.addToLpdb = CustomLeague.addToLpdb
+	league.liquipediaTierHighlighted = CustomLeague.liquipediaTierHighlighted
+	league.appendLiquipediatierDisplay = CustomLeague.appendLiquipediatierDisplay
+
+	return league:createInfobox()
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets, Cell{
+		name = 'Teams',
+		content = {_args.team_number}
+	})
+
+	return widgets
+end
+
+function CustomLeague:liquipediaTierHighlighted(args)
+	return Logic.readBool(args['supercell-sponsored'])
+end
+
+function CustomLeague:appendLiquipediatierDisplay()
+	return Logic.readBool(_args['supercell-sponsored']) and ('&nbsp;' .. SUPERCELL_SPONSORED_ICON) or ''
+end
+
+function CustomLeague:addToLpdb(lpdbData, args)
+	lpdbData.publishertier = Logic.readBool(_args['supercell-sponsored'])
+	lpdbData.participantsnumber = args.team_number
```

Comment:
1) wouldn't we want to force this to be numeric?
2) doesn't commons already set it?

Reply:
Assuming this refers to  `lpdbData.participantsnumber = args.team_number`
If the line is removed, the lpdb data under amount of teams became **-1** 

I assume we need to keep this line but rewrite in a way that it can only be numbers?

Reply:
forcing team_number to always be numeric would be a good idea, but other infoLeague didnt do much of this too

as for 2.  I did try removed **lpdbData.participantsnumber = args.team_number**
the stored number in **participantsnumber** then jumps back to -1 regardless of the input in |team_number

Reply:
Since #3098 this should no longer be needed